test/arrayops.jl

```diff
@@ -2133,6 +2133,11 @@ Base.convert(::Type{Float64}, x::F21666) = Float64(x.x)
     @test norm(f_truth - f_rounds) < norm(f_truth - f_inexact)
 end
 
+struct OnesUsesOneunit
+    val::Int16
+end
+Base.one(::Type{OnesUsesOneunit}) = OnesUsesOneunit(1).val
+Base.zero(::Type{OnesUsesOneunit}) = OnesUsesOneunit(0)
```

Comment:
Instead of defining your own type here, just use the `Furlong` type from `tests/dimensionful.jl`, which is there precisely to add tests for dimension bugs.